.gitlab/README.md

```diff
@@ -0,0 +1,100 @@
+                    Finite Element Discretization Library
+                                   __
+                       _ __ ___   / _|  ___  _ __ ___
+                      | '_ ` _ \ | |_  / _ \| '_ ` _ \
+                      | | | | | ||  _||  __/| | | | | |
+                      |_| |_| |_||_|   \___||_| |_| |_|
+
+                               https://mfem.org
+
+
+This directory contains most of the GitLab CI configuration. MFEM runs both PR
+and nightly testing on GitLab.
+
+# Structure
+
+## Top level
+
+The root configuration file is `.gitlab-ci.yml` at the root of MFEM repo.
+This file only defines one stage, in which we trigger several
+sub-pipelines.
+
+We use sub-pipelines to isolate the test for one combination of `machine`
+and `test type`.
+
+Machines typically include:
+
+* Quartz: Intel bi-socket x86
+* Lassen: Power9 + Nvidia GPU
+* Corona: AMD GPU
+
+Test types include:
+
+* Build and test: Spack driven build of dependencies, mfem build, mfem
+  test
+* Baseline: Script driven build of dependencies, thorough testing
+
+**NOTE**
```

Comment:
Maybe use a warning emoji ⚠️ instead of **NOTE**?

⚠️ The sub-pipeline design allows to add a new machine or a new test type without
altering the scheduling, execution and displaying of the others.

Reply:
An icon would certainly be nice, not sure about a "warning" in that case.